src/shogun/features/DenseFeatures.cpp

```diff
@@ -435,6 +404,18 @@ template<class ST> bool CDenseFeatures<ST>::apply_preprocessor(bool force_prepro
 	}
 }
 
+template<class ST> void CDenseFeatures<ST>::eval_features()
+{
+	auto first_vec = get_feature_vector(0);
+	SGMatrix<ST> target(first_vec.vlen, get_num_vectors());
+	target.set_column(0, first_vec);
+
+	for (index_t i = 1; i < get_num_vectors(); ++i)
```

Comment:
this whole part above should be rather done by a smart memcpy instead of setting the columns separately. needs a stride etc.